yellowbrick/features/histogram.py

```diff
@@ -0,0 +1,202 @@
+# For license information, see LICENSE.txt
+#
+# ID: histogram.py
+
+"""
+Implements histogram with vertical lines to help with balanced binning.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+import matplotlib.pyplot as plt
+import numpy as np
+
+from yellowbrick.features.base import FeatureVisualizer
+from yellowbrick.exceptions import YellowbrickValueError
+from yellowbrick.utils import is_dataframe
+
+##########################################################################
+## Balanced Binning Reference
+##########################################################################
+
+class BalancedBinningReference(FeatureVisualizer):
+    """
+    BalancedBinningReference allows to generate a histogram with vertical lines
+    showing the recommended value point to bin your data so they can be evenly
+    distributed in each bin.
+
+    Parameters
+    ----------
+    ax : matplotlib Axes, default: None
+        This is inherited from FeatureVisualizer and is defined within
+        BalancedBinningReference.
+    feature : string, default: None
+        The name of the X variable
+        If a DataFrame is passed to fit and feature is None, feature
+        is selected as the column of the DataFrame.  There must be only
+        one column in the DataFrame.
+    target : string, default: None
+        The name of the Y variable
+        Default: Counts
+    bins : number of bins to generate the histogram
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Examples
+    --------
+    >>> visualizer = BalancedBinningReference()
+    >>> visualizer.fit(X)
+    >>> visualizer.poof()
+
+
+    Notes
+    -----
+    These parameters can be influenced later on in the visualization
+    process, but can and should be set as early as possible.
+    """
+
+    def __init__(self, ax=None, feature=None, target='Counts', bins=4, **kwargs):
+
+        super(BalancedBinningReference, self).__init__(ax, **kwargs)
+
+        self.feature = feature
+        self.target = target
+        self.bins = bins
+
+    def draw(self, X, **kwargs):
+        """
+        Draws a histogram with the reference value for binning as vetical lines
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x 1
+            A matrix of n instances with 1 feature
+        """
+
+        # draw the histogram
+        hist, bin_edges = np.histogram(X, bins=self.bins)
+        self.ax.hist(X, bins=self.bins, color=kwargs.pop("color", "#6897bb"), **kwargs)
+
+        # add vetical line with binning reference values
+        plt.vlines(bin_edges,0,max(hist),colors=kwargs.pop("colors", "r"))
+
+        #print the binning reference values
+        print("The binning reference values are:", bin_edges)
```

Comment:
Please remove the debug print statement for library code, we don't know what context the user will be running yellowbrick in, and we don't want to write anything to stdout. If this is information the user might want (and I think they would), I would recommend storing the values in `self.bin_edges_` -- a learned attribute of the estimator.